engine/models_internal.go

```diff
@@ -97,10 +97,26 @@ func (m *Model) SetRecordTime(created bool) {
 // UpdateMetadata will update the metadata on the model
 // any key set to nil will be removed, other keys updated or added
 func (m *Model) UpdateMetadata(metadata Metadata) {
```

Comment:
Why don't use a "normal" assigmnent?

A deep copy or shallow copy would only be necessary if the incoming metadata were modified elsewhere after being passed. However, in our case—as far as I can tell—there is no such scenario.

Please check if my solution works:

```golang
func (m *Model) UpdateMetadata(metadata Metadata) {
	m.Metadata = metadata
}
```

Reply:
It doesn't handle case of cleared all metadata when given an empty input 
I've got this error: `panic recovered: assignment to entry in nil map` If I want to change a metadata after deleting

Reply:
So? Just handle the empty-map case.

Reply:
Updated

Reply:
I tested my solution on my own. It worked on my side.
When have you got the `panic recovered` error. 

Did you copy paste my code:
```golang
func (m *Model) UpdateMetadata(metadata Metadata) {
	m.Metadata = metadata
}
```
which you should - or you've added some lines below?

Reply:
With line above. Added your solution